lib/java/src/org/apache/thrift/transport/TFastFramedTransport.java

```diff
@@ -106,8 +107,8 @@ public TFastFramedTransport(TTransport underlying, int initialBufferCapacity, in
     this.underlying = underlying;
     this.maxLength = maxLength;
     this.initialBufferCapacity = initialBufferCapacity;
-    writeBuffer = new AutoExpandingBufferWriteTransport(initialBufferCapacity);
     readBuffer = new AutoExpandingBufferReadTransport(initialBufferCapacity);
+    writeBuffer = new AutoExpandingBufferWriteTransport(initialBufferCapacity);
```

Comment:
I suppose it's impossible (or not recommended) to `write(sizeFiller, 0, 4)` here?

Rationale: we do the special case once, and avoid doing a conditional on every `write()` call. Since this call doesn't hit an IO resource it *shouldn't* throw exceptions either.

Reply:
If the only class consuming AutoExpandingBufferWriteTransport is TFastFramedTransport, we could embed the reservation for the size into that.  Thoughts?